pkg/agent/testdata/AKSUbuntu1604+CustomKubeletConfig+CustomLinuxOSConfig/line33.sh

```diff
@@ -97,7 +97,9 @@ ensureMonitorService
 # must run before kubelet starts to avoid race in container status using wrong image
 # https://github.com/kubernetes/kubernetes/issues/51017
 # can remove when fixed
-cleanupRetaggedImages
+if [[ "AzurePublicCloud" == "AzureChinaCloud" ]]; then
```

Comment:
basic question: isn't this check only doing a string comparison that always evaluates to false? I went through the link in the comment but I'm probably missing something about this check.

Reply:
you are looking at an autogenerated file, line 33.sh
these test files essentially act as what the cse will look like for certain NodeBootstrappingConfigurations(in this case, it is AKSUbuntu1604+CustomKubeletConfig+CustomLinuxOSConfig) (look at the file path)
The actual check is in cse_main.sh 
if [[ "{{GetTargetEnvironment}}" == "AzureChinaCloud" ]]; then
    retagMCRImagesForChina
fi

so what the check essentially does is if the target environment is AzureChinaCloud, then the retag function is called. For the file you commented on, the target env is AzurePublicCloud, as expected from the test configuration,

Reply:
Thanks for explaining this @anujmaheshwari1! So `GetTargetEnvironment` would be replaced depending on the environment in which it is run in.